hawkmoth/parser.py

```diff
@@ -239,10 +239,29 @@ def _recursive_parse(comments, cursor, nest, compat):
 
     return [(doc, meta)]
 
+def extract_filename_line(source_loc):
+    valid = re.compile(r".*file\s'(.*)',\sline\s(\d+).*")
+    match = valid.match(source_loc)
+    filename = ''
+    line = -1
+    if match:
+        filename = match.group(1)
+        line = int(match.group(2))
+
+    return filename, line
+
+def parse_gather_diagnostics(severity, errors, diagnostics):
+    for diag in diagnostics:
+        if diag.severity >= severity:
+            filename, line = extract_filename_line(str(diag.location))
```

Comment:
Use `diag.location.file.name` and `diag.location.line` (if I remember correctly). No need for importing `re` either.

Reply:
Yep, with that it looks much more cleaner in your version.